npk/src/npk.rs

```diff
@@ -755,113 +796,123 @@ fn create_squashfs_img(
         .arg("-info")
         .arg("-pf")
         .arg(pseudo_files.path());
-
     if let Some(block_size) = squashfs_opts.block_size {
         cmd.arg("-b").arg(format!("{}", block_size));
     }
-
-    cmd.output()
-        .map_err(|e| Error::Squashfs(format!("Failed to execute '{}': {}", &MKSQUASHFS_BIN, e)))?;
-
-    if !src.exists() {
-        Err(Error::Squashfs(format!(
-            "'{}' did not create '{}'",
-            &MKSQUASHFS_BIN, &FS_IMG_NAME
-        )))
-    } else {
+    task::block_in_place(|| -> Result<(), Error> {
+        cmd.output().map_err(|e| {
+            Error::Squashfs(format!("Failed to execute '{}': {}", &MKSQUASHFS_BIN, e))
+        })?;
+        if !src.exists() {
+            return Err(Error::Squashfs(format!(
+                "'{}' did not create '{}'",
+                &MKSQUASHFS_BIN, &FS_IMG_NAME
+            )));
+        }
         Ok(())
-    }
+    })?;
+    Ok(())
 }
 
-fn unpack_squashfs(image: &Path, out: &Path) -> Result<(), Error> {
-    if which::which(&UNSQUASHFS_BIN).is_err() {
+async fn unpack_squashfs(image: &Path, out: &Path) -> Result<(), Error> {
+    let squashfs_root = out.join("squashfs-root");
+
+    if task::block_in_place(|| which::which(&UNSQUASHFS_BIN).is_err()) {
         return Err(Error::Squashfs(format!(
             "Failed to locate '{}'",
             &UNSQUASHFS_BIN
         )));
     }
-    if !image.exists() {
+    if !task::block_in_place(|| image.exists()) {
```

Comment:
Think this `block_in_place` is not needed.

Reply:
Isn't checking for file existence considered a slow IO operation?